department-of-veterans-affairs/caseflow
# hide ihp tasks on substitute if poa cannot write ihp tasks

Resolves [CASEFLOW-1627](https://vajira.max.gov/browse/CASEFLOW-1627)
Resolves #16431

### Description
This adds a serializer for POAs, along with a variable, `ihp_allowed`, to be used in the frontend.  It also adjust the frontend logic to use that new variable.  The new variable checks the type of organization on the poa to see whether the new claimant's representative organization can write ihps.  Specifically, Field Vsos and Private Bars cannot write IHPs, so this makes it so the COB user won't see IHPs listed at substitution task-selection time, once they select a claimant with a Field VSO or Private Bar organization representing them.
I'm open to suggestions on where the organization-consulting logic can live.  I thought it best to put it into the serializer because it's only needed in a frontend context, but I would agree that it's somehow wrong to be querying the db there.  It could be a method on Bgs_power_of_attorney, rather.
**TO DO:**
- [ ] optimize db call for this if possible

### Acceptance Criteria
- [X] New unit tests for added backend logic

### Testing Plan
updated unit tests.  To view the change in action:
1. Create the data situation:  use the new `with_ihp_task` trait to create an appeal, then mark the ihp task as completed:
```
RequestStore[:current_user] = User.system_user
appeal = FactoryBot.create(:appeal, :with_ihp_task, :dispatched_with_decision_issue,
         docket_type: "direct_review",
         stream_type: "original",
         disposition: "dismissed_death",               
         veteran_is_not_claimant: false)
appeal1.veteran.update(date_of_death: 1.day.ago) 
appeal.uuid
```
2. Navigate to caseflow and become a COB user, then navigate to the appeal you created at `http://localhost:3000/queue/appeals/<appeal-uuid>`
3. Click the button to add a substitute appellant
4. Select the first relationship, check the network tab and click on the `poa` call.  You should see the value for `ihp_allowed` match the display: if it's true, that task will be available, and if false, it will not.  Available relationships include all three potential setups: ihp-writing poa, non-ihp-writing-poa, and no poa.  No poa is currently configured to be non-ihp-writing, meaning that substitute appellants are not expected to represent themselves in an informal hearing presentation.
